src/decawave/src/decawave_node.cpp

```diff
@@ -110,29 +85,28 @@ void update_callback(const ros::TimerEvent&){
   decawave::Range msg;
 
   // update decawave data
-  piTag->updateSamples();
-
-  // get tag position from the decawave
-  tagPos = piTag->getPos();
-
-  ros::param::get("~port_num",port_num);
-  std::string frame_id = std::to_string(port_num);
-
-  msg.distance = tagPos.x;
-  msg.estimated_variance = 0.10;
-  // fill out message header
-  msg.header.stamp = ros::Time::now();
-  msg.header.frame_id = "decawave" + frame_id;//"decawave_" + port_num;
-  msg.child_frame_id = "decawave2_link"; //change to correct part
-
-  gps_pub.publish(msg);
-
-  msg.distance = tagPos.y;
-  msg.estimated_variance = 0.10;
-  // fill out message header
-  msg.header.stamp = ros::Time::now();
-  msg.header.frame_id = "decawave" + frame_id;//"decawave_" + port_num;
-  msg.child_frame_id = "decawave3_link"; //change to correct part
-
-  gps_pub.publish(msg);
+  std::vector<anchor> anchors = piTag->updateSamples();
+
+  ros::param::get("~port_num",port_name);
+  std::string frame_id = port_name;
+
```

Comment:
No no no. 
Params are global, usually only written in the int main, and read-only thereafter.